app/src/main/java/me/phh/treble/app/Lenovo.kt

```diff
@@ -0,0 +1,43 @@
+package me.phh.treble.app
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.media.AudioManager
+import android.os.SystemProperties
+import android.os.UEventObserver
+import android.preference.PreferenceManager
+import android.util.Log
+import java.io.File
+
+object Lenovo: EntryStartup {
+    val dtPanel = "/sys/devices/virtual/touch/tp_dev/gesture_on" //K5P
```

Comment:
Do you have reasons to believe this is Lenovo specific, but that all Lenovo devices have that?

Reply:
I think Lenovo devices have that, such as Z5 and s5pro, but I'm not sure. I only have Lenovo K5 pro.
Generic.kl must add key 542 wakeup

Reply:
I tested it with Lenovo k5pro,support that.

Reply:
Ok, so in LenovoSettings.kt's enabled() please check for dtPanel file existence.

For `key 542`, first step is to know what's the name of the input that triggers it (changing Generic.kl isn't acceptable). To do that, run `adb shell getevent` from a computer, and do the dt2w, then from the logs, determine which /dev/input/eventX triggers the event, and also from logs of the getevent determine what's the name of the input device. (I recommend you attach logs of `getevent` and `dumpsys input`). Then you need to do something like that: https://github.com/phhusson/device_phh_treble/commit/954d2157da27a54e3a711e0331a90215c6a9c4b3

I can do that last part for you, just give me the getevent and dumpsys results

Reply:
kunlun:/ $ getevent
add device 1: /dev/input/event4
  name:     "uinput-fpc"
add device 2: /dev/input/event3
  name:     "Headset"
add device 3: /dev/input/event0
  name:     "qpnp_pon"
add device 4: /dev/input/event1
  name:     "synaptics_dsx"
could not get driver version for /dev/input/mouse0, Not a typewriter
could not get driver version for /dev/input/mice, Not a typewriter
add device 5: /dev/input/event2
  name:     "gpio-keys"
/dev/input/event1: 0001 021e 00000001
/dev/input/event1: 0000 0000 00000000
/dev/input/event1: 0001 021e 00000000
/dev/input/event1: 0000 0000 00000000

Reply:
5: synaptics_dsx
      Classes: 0x00000015
      Path: /dev/input/event1
      Enabled: true
      Descriptor: 954faadc99bb5a7c1d0537b923e0490c90b47e98
      Location: synaptics_dsx/touch_input
      ControllerNumber: 0
      UniqueId: 
      Identifier: bus=0x0000, vendor=0x0000, product=0x0003, version=0x2070
      KeyLayoutFile: /system/usr/keylayout/synaptics_dsx.kl
      KeyCharacterMapFile: /system/usr/keychars/Generic.kcm
      ConfigurationFile: 
      HaveKeyboardLayoutOverlay: false


    KeyEvent(deviceId=5, source=0x00000101, action=DOWN, flags=0x00000008, keyCode=224, scanCode=542, metaState=0x00000000, repeatCount=0), policyFlags=0x02000000, age=1176314.8ms
    KeyEvent(deviceId=5, source=0x00000101, action=UP, flags=0x00000008, keyCode=224, scanCode=542, metaState=0x00000000, repeatCount=0), policyFlags=0x02000000, age=1176314.8ms
    KeyEvent(deviceId=4, source=0x00000101, action=DOWN, flags=0x00000008, keyCode=26, scanCode=116, metaState=0x00000000, repeatCount=0), policyFlags=0x22000000, age=12101.3ms
    KeyEvent(deviceId=4, source=0x00000101, action=UP, flags=0x00000008, keyCode=26, scanCode=116, metaState=0x00000000, repeatCount=0), policyFlags=0x22000000, age=11998.7ms
    KeyEvent(deviceId=5, source=0x00000101, action=DOWN, flags=0x00000008, keyCode=224, scanCode=542, metaState=0x00000000, repeatCount=0), policyFlags=0x02000000, age=10377.0ms
    KeyEvent(deviceId=5, source=0x00000101, action=UP, flags=0x00000008, keyCode=224, scanCode=542, metaState=0x00000000, repeatCount=0), policyFlags=0x02000000, age=10376.9ms

Reply:
Lenovo K5pro 
ro.vendor.build.fingerprint=Lenovo/kunlun/kunlun:9/PKQ1.190127.001/11.1.099_191008:user/release-keys